compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -59,15 +59,30 @@ impl<'a> Parser<'a> {
         self.current_closure.take();
 
         let attrs = self.parse_outer_attributes()?;
-        self.parse_expr_res(Restrictions::empty(), attrs)
+        self.parse_expr_res(Restrictions::empty(), attrs).map(|res| res.0)
     }
 
     /// Parses an expression, forcing tokens to be collected.
     pub fn parse_expr_force_collect(&mut self) -> PResult<'a, P<Expr>> {
         self.current_closure.take();
 
+        // If the expression is associative (e.g. `1 + 2`), then any preceding
```

Comment:
Attributes have the priority of an unary operator (like `-`, `!`, `*`, `&`, `box`), or at least they are supposed to.
So these interactions may affect not only binary operators (including assignments), but also ranges `#[attr] a .. b` (binds to `a`), casts `#[attr] a as Type` (binds to `a`), maybe something else.

I'm not sure, is all that covered by "expression is associative"?

Reply:
I think so. `AssocOp::from_token` includes `..`, `..=`, `...`, and `as`.